util.go

```diff
@@ -5,18 +5,29 @@ import (
 	crand "crypto/rand"
 	"encoding/binary"
 	"fmt"
+	"math/big"
 	"math/rand"
 	"time"
 
 	"github.com/hashicorp/go-msgpack/codec"
 )
 
+var rnd = rand.New(rand.NewSource(newSeed()))
+
+func newSeed() int64 {
```

Comment:
Can you add a doc string and a test to verify that the first 1000 results are unique.